kbfsgit/runner.go

```diff
@@ -1439,20 +1439,26 @@ func (r *runner) pushAll(ctx context.Context, fs *libfs.FS) (err error) {
 // not in `remoteStorer`.
 func (r *runner) parentCommitsForRef(ctx context.Context,
 	localStorer gogitstor.Storer, remoteStorer gogitstor.Storer,
-	refs map[string]bool) (libgit.CommitsByRefName, error) {
+	refs map[gogitcfg.RefSpec]bool) (libgit.RefDataByName, error) {
 
-	commitsByRef := make(libgit.CommitsByRefName, len(refs))
+	commitsByRef := make(libgit.RefDataByName, len(refs))
 	haves := make(map[plumbing.Hash]bool)
 
-	for refName := range refs {
-		ref, err := localStorer.Reference(plumbing.ReferenceName(refName))
+	for refspec := range refs {
+		if refspec.IsDelete() {
```

Comment:
It would be pretty weird for `pushAll` to have a delete refspec, but I guess it's not impossible.  Anyway, this code isn't too intrusive I guess.